packages/runtime/container-runtime/src/summarizerHeuristics.ts

```diff
@@ -0,0 +1,95 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Timer } from "@fluidframework/common-utils";
+import { ISummaryConfiguration } from "@fluidframework/protocol-definitions";
+import { ISummarizeHeuristicData, ISummarizeHeuristicRunner, ISummaryAttempt } from "./summarizerTypes";
+import { SummarizeReason } from "./summaryGenerator";
+
+/** Simple implementation of class for tracking summarize heuristic data. */
+export class SummarizeHeuristicData implements ISummarizeHeuristicData {
+    protected _lastAttempt: ISummaryAttempt;
+    public get lastAttempt(): ISummaryAttempt {
+        return this._lastAttempt;
+    }
+
+    protected _lastSuccessfulSummary: Readonly<ISummaryAttempt>;
+    public get lastSuccessfulSummary(): Readonly<ISummaryAttempt> {
+        return this._lastSuccessfulSummary;
+    }
+
+    constructor(
+        public lastOpSequenceNumber: number,
+        attemptBaseline: ISummaryAttempt,
```

Comment:
I am a little confused about the purpose of `attemptBaseline`. This basically is the sequence number of the delta manager when summarizer was created and the creation time. What role does that have to play with the heuristics?

Reply:
Looking at `ISummaryAttempt`, it seems that this should be set when there is an actual summary attempt. But creating the summarizer and the associated class doesn't mean we are attempting a summary right.

Reply:
It's the baseline attempt data used for comparisons with subsequent attempts / calculations.

Reply:
Basically, this is what to use when there is no prior attempt data, or it isn't yet known.

Reply:
Okay, that makes sense. Can you please add a comment here explaining this?

Reply:
Also added a comment there